### PR TITLE
Function cast support Spark text output

### PR DESCRIPTION
**Pull request by zhanglistar**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
 The output of function `cast` differs with Apache Spark which cause difference in gluten project, see https://github.com/apache/incubator-gluten/issues/7602
This PR adds Spark text output format support feature, default closed. 

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit 24a3d1574d9ce15337e9234dfab2780ada6eba68 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/70957/24a3d1574d9ce15337e9234dfab2780ada6eba68/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/70938/ff191b0b089af03d2134dd1f2bedd390cf42482f/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: src/Core/FormatFactorySettings.h [hunk 1]
```diff
@@ -1242,6 +1242,9 @@ Set the quoting rule for identifiers in SHOW CREATE query
 )", 0) \
     DECLARE(IdentifierQuotingStyle, show_create_query_identifier_quoting_style, IdentifierQuotingStyle::Backticks, R"(
 Set the quoting style for identifiers in SHOW CREATE query
+)", 0) \
+    DECLARE(Bool, spark_text_output_format, false, R"(
```
Comment (vdimir) [inline src/Core/FormatFactorySettings.h:1246]: Maybe we can introduce it in a bit more generic way, like `composed_data_type_output_format_mode = 'default'/'spark'` so we can support more modes in the future and not making `spark` special case

Comment (zhanglistar) [reply src/Core/FormatFactorySettings.h:1246]: Good point. I will try later.

File: src/Core/SettingsChangesHistory.cpp [hunk 2]
```diff
@@ -141,6 +141,7 @@ static std::initializer_list<std::pair<ClickHouseVersion, SettingsChangesHistory
             {"allow_reorder_prewhere_conditions", true, true, "New setting"},
             {"input_format_parquet_bloom_filter_push_down", false, true, "When reading Parquet files, skip whole row groups based on the WHERE/PREWHERE expressions and bloom filter in the Parquet metadata."},
             {"date_time_64_output_format_cut_trailing_zeros_align_to_groups_of_thousands", false, false, "Dynamically trim the trailing zeros of datetime64 values to adjust the output scale to (0, 3, 6), corresponding to 'seconds', 'milliseconds', and 'microseconds'."},
+            {"composed_data_type_output_format_mode", "default", "default", "New setting"},
```
Comment (vdimir) [inline src/Core/SettingsChangesHistory.cpp:144]: Let's move it to `24.12`

**Review by vdimir — APPROVED** [review]

File: src/Core/SettingsChangesHistory.cpp (hunk 2 above)
Comment (zhanglistar) [reply src/Core/SettingsChangesHistory.cpp:144]: done

**Review by vdimir — APPROVED** [review]
Thanks for taking the time to improve the project!

**Comment by zhanglistar** [comment]
> Thanks for taking the time to improve the project!

We are heavily use Clickhouse in Gluten prject and as a OLAP server on our business. It's my honor to contribute.

**Pull request merged**